api/src/functions/otel-trace/otel-trace.ts

```diff
@@ -264,13 +264,35 @@ async function createSpan(span: Span, resourceId: string, scopeId: string) {
   })
 }
 
-export const handler = async (event: APIGatewayEvent, _context: Context) => {
-  // Some guidance is available from:
-  // https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md
-  // Note the spec for OTLP traces is here:
-  // https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto
+let spanProcessingInterval: NodeJS.Timeout | null = null
+const spanDataQueue: string[] = []
+export const startSpanProcessor = async () => {
+  logger.info('Starting OpenTelemetry span processor')
+
+  if (spanProcessingInterval) {
+    clearInterval(spanProcessingInterval)
+  }
+
+  const processSpanBatch = async () => {
+    if (spanDataQueue.length === 0) {
+      return
+    }
+    const spans = spanDataQueue.splice(0, spanDataQueue.length)
+    for (let i = 0; i < spans.length; i++) {
+      await processSpan(spans[i])
```

Comment:
Would it be even better to do these all in parallell rather than sequential?

Reply:
Yeah good point! Updated in [6a0cc71](https://github.com/redwoodjs/studio/pull/47/commits/6a0cc71298d1b29c513d75dbdefd075ac3be8108) 

There are loads of `for` loops in here - I'm a fan of old school loops haha - that might benefit from the same treatment. I'd want to do that separately as part of a fully measure changed of the ingestions to be more performant.

Reply:
> I'd want to do that separately

Sounds good to me :)